### DISCUSSION TITLE
New symlink for cores that are using a .mgl file on MiSTer.

### DISCUSSION THREAD

**Discussion by toaks2804** [retronas/discussions#10] [Q&A] [answered]
Hello. I am using mgl files for the NeoGeo CD and PSX2XCPU cores. I am using them so I can have seperate configurations saved for these cores that don't affect the base core, but I want them to share the same roms folders. I am placing all my roms into the roms directories on Retronas. How would I make new symlinks to point to the correct roms folders, and how would I later remove them if I decide to change my setup? Thanks for any help. I know there was a similar subject about the genesis core, but this is a little different I think because of the use of mgl files?

**Comment by sairuk** [comment]
you create a link with the `ln` command, a soft link with `-s`. Have a look at `man ln`  in the terminal, it will get you to the manual page then you point whatever source to whatever destination. to remove items later see the `rm` command manual.

There  isn't exactly enough info there to suggest anymore of a solution but good luck.

**Comment by toaks2804** [comment]
Thanks for the reply. Heres a example. I have the standard PSX core files in the media/fat/games/PSX directory on the mister and it shows the rom files from retronas/roms/sony/playstation1. I have the PSX2XCPU in its own folder at media/fat/games/PSX2XCPU. I want it to show the roms from the retronas/roms/sony/playstation1 directory as well. And I also need a command to remove the link if necessary. I'm not very experienced with Linux so I am very unfamiliar with terminal commands.Also, would I run these commands on the MiSTer or on the Raspberry pi that is running RetroNAS? Thanks again!

**Reply by sairuk** [reply, marked as answer]
you would do something like this on retronas

creation
```
cd /data/retronas/mister/games
mv PSX2XCPU PSX2XCPU_old
ln -s ../../roms/sony/playstation1 PSX2XCPU
```

removal
```
cd /data/retronas/mister/games
rm PSX2XCPU
```

**Reply by sairuk** [reply]
you'd also need to repeat the process for saves, BIOS etc if they have their own folders for that core

**Reply by toaks2804** [reply]
Thank you. I'll give it a try! I think I am understanding sort of how it works. I've also looked at a guide on Symlinks. What does the line: mv PSX2XCPU PSX2XCPU_old do? I can't find a reference for that command. I really appreciate all the help. Trying to understand better how this all works

**Reply by sairuk** [reply]
its only there to move the old folder (if it exists) to a different name so the symlink can be created

**Reply by toaks2804** [reply]
OK, I think I almost have it working. The problem I'm having is it is creating a symlink to a file and not a directory. also the path of data/retronas/mister/games was not working. Mine are in storage/mister/games and storage/roms. I am running the command on the terminal on my pi4. This is what I tried:

cd /storage/mister/games
ln -s /storage/roms/atari/jaguar Jaguar

it then creates a file named Jaguar instead of a directory. Any ideas?

**Reply by sairuk** [reply]
technically a link is a file, when you say it creates a file instead of a directory is that how it looks from the share?

if so, that probably means the link is broken (source doesn't exist), the commands you have listed are technically correct providing your paths are accurate.

the link creation command will be incorrect resulting in a broken link

**Reply by toaks2804** [reply]
This is what I get when it runs that command:
![Screenshot 2023-10-06 at 1 44 32 PM](https://github.com/danmons/retronas/assets/37721294/ab17a9ec-48ed-42b1-befc-bc99ca663236)

**Reply by sairuk** [reply]
ahhh i see, I wonder why, lets see if it was created correctly

on the terminal

check the link is pointing to the correct location
```
file /storage/mister/games/Jaguar
```

then double check the source dir is a dir
```
file /storage/roms/atari/jaguar
```

**Reply by toaks2804** [reply]
the first line replied with this: /storage/mister/games/Jaguar: broken symbolic link to storage/roms/atari/jaguar
the second replied with this: /storage/roms/atari/jaguar: directory

**Reply by sairuk** [reply]
you forgot the starting forward slash when you created the symlink which is evident here in the output

```
to storage/roms/atari/jaguar
```

**Reply by sairuk** [reply]
so instead of linking from the top of the tree (/) it is linked to a non-existent sub directory of mister/games

**Reply by toaks2804** [reply]
I absolutely did. That fixed it. Thanks so much!!!
